packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -43,6 +43,11 @@ export default class RequestClient {
     this.onReceiveResponse = this.onReceiveResponse.bind(this)
     this.allowedHeaders = ['accept', 'content-type', 'uppy-auth-token']
     this.preflightDone = false
+    this.companionHeaders = opts !== undefined && opts.companionHeaders !== undefined ? opts.companionHeaders : {}
+  }
+
+  setCompanionHeaders (headers) {
+    this.companionHeaders = headers
   }
```

Comment:
Same here, we should not make it publicly available. You can add a `[Symbol.for('uppy test: getCompanionHeaders')]` method to access it from the tests.
```suggestion
    this.#companionHeaders = headers
  }
  
  [Symbol.for('uppy test: getCompanionHeaders')] () { return this.#companionHeaders }
```